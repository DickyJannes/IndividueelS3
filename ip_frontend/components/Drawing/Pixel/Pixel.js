/* eslint-disable react/prop-types */
import React, {useState} from "react";
import { View } from "react-native";

//import pixelStyles from "../styles/pixel";

export default function Pixel(props) {
	const {selectedColor, newBackground} = props;

	const [pixelColor, setPixelColor] = useState(newBackground);
	const [oldColor, setOldColor] = useState(pixelColor);
	const [canChangeColor, setCanChangeColor] = useState(true);

	function applyColor() {
		setPixelColor(selectedColor);
		setCanChangeColor(false);
	}
    
	function changeColorOnHover() {
		setOldColor(pixelColor);
		setPixelColor(selectedColor);
        
	}
    
	function resetColor() {
		if (canChangeColor) {
			setPixelColor(oldColor);
		}
    
		setCanChangeColor(true);
	}
  
	return (
		<View style={{backgroundColor: pixelColor, 
			width: "1.5rem", 
			height: "1.5rem", 
			opacity: 0.9,
			borderColor: "gray",
			borderWidth: 0,
			borderRadius: 0}}
		onClick={applyColor}
		onMouseEnter={changeColorOnHover}
		onMouseLeave={resetColor}
		></View>
	);
}