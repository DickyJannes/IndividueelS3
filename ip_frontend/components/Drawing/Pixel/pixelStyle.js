import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
	pixel: {
		width: "1.5rem",
		height: "1.5rem",
		borderColor: "red",
		borderWidth: 1,
	}
});

export default styles;
