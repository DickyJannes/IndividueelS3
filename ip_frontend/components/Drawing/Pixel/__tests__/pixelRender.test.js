import React from 'react';
import renderer from 'react-test-renderer';

import Pixel from '../Pixel';

test('renders pixel correctly', () => {
  const tree = renderer.create(<Pixel />).toJSON();
  expect(tree).toMatchSnapshot();
});