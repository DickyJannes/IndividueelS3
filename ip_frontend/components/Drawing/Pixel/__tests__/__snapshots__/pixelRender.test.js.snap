// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pixel correctly 1`] = `
<View
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderColor": "gray",
      "borderRadius": 0,
      "borderWidth": 0,
      "height": "1.5rem",
      "opacity": 0.9,
      "width": "1.5rem",
    }
  }
/>
`;
